### PR TITLE
Using LDK guide

### LINKED ISSUE #32: Add "Using LDK" guide
- [ ] remove "Using LDK" section from main "Build a Node" guide
- [ ] add separate guide with:
* opening a channel (public + private)
* [force]closing a channel
* sending a payment to an invoice
* list [usable] channels
* list peers
* get node pubkey
* get invoice
* connect peer

Should be the same format as "Build a Node" guide w/ Rust code samples taken from the sample + modified to suit the docs, but should also the existing docs Java samples, especially Matt's ["Opening a Channel" Java sample](https://lightningdevkit.org/docs/build_node/#opening-a-channel).

### PR DESCRIPTION
**Pull request by jkczyz**
Add a "Using LDK" guide, split out from the guide for building a node. Closes #32.

Includes text and examples in Rust and Java. The Java examples were largely copied over but should be updated once the Rust examples are settled.

### CODE REVIEW

**Comment by jkczyz** [comment]
@valentinewallace Noticed the examples for opening a channel are inconsistent between languages. Will update them but let me know if you catch anything else.

File: docs/using_ldk.md [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+        NetworkGraph graph = netgraph.graph();
+        Result_RouteLightningErrorZ route_res = UtilMethods.get_route(chan_manager.get_our_node_id(),
+            graph, invoice.recover_payee_pub_key(), invoice.features(),
+            channel_manager.list_usable_channels(), invoice.route_hints(), amt,
+            invoice.min_final_cltv_expiry(), logger);
+        assert route_res instanceof Result_RouteLightningErrorZ.Result_RouteLightningErrorZ_OK;
+        route = ((Result_RouteLightningErrorZ.Result_RouteLightningErrorZ_OK) route_res).res;
+    }
+
+    Result_NonePaymentSendFailureZ payment_res = peer1.chan_manager.send_payment(route, invoice.payment_hash(),
```
Comment (valentinewallace) [inline docs/using_ldk.md:380]: Oops, to be consistent: s/peer1.chan_manager/channel_manager

File: docs/using_ldk.md [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+        amt = ((Option_u64Z.Some)invoice.amount_pico_btc()).some;
+    }
+    if (amt == 0) {
+        // <Handle a zero-value invoice>
+    }
+
+    Route route;
+    try (LockedNetworkGraph netgraph = router.read_locked_graph()) {
+        NetworkGraph graph = netgraph.graph();
+        Result_RouteLightningErrorZ route_res = UtilMethods.get_route(chan_manager.get_our_node_id(),
```
Comment (valentinewallace) [inline docs/using_ldk.md:372]: for consistency: s/chan_manager/channel_manager

File: docs/using_ldk.md [hunk 3]
```diff
@@ -0,0 +1,534 @@
...
+
+</TabItem>
+<TabItem value="java">
+
+```java
+String description = "coffee";
+long amt_msat = 10_000L;
+Result_InvoiceSignOrCreationErrorZ invoice = UtilMethods.invoice_from_channelmanager(
+    channel_manager, keys_manager.as_KeysInterface(), LDKCurrency.LDKCurrency_Bitcoin,
+	Option_u64Z.some(amt_msat), "coffee");
```
Comment (valentinewallace) [inline docs/using_ldk.md:471]: I think the `description` var is meant to be used on line 471

**Review by valentinewallace — COMMENTED** [review]
Last few nits, then feel free to squash and I'll merge!

**Pull request merged**